src/vivarium_cluster_tools/vadmin/repositories.py

```diff
@@ -220,11 +213,31 @@ def clone_repository(repository_url: str, output_dir: Path) -> Path:
 
 
 def get_library_versions():
-    libraries = ['vivarium', 'vivarium_public_health', 'vivarium_cluster_tools']
+    libraries = ['vivarium', 'vivarium_public_health', 'vivarium_cluster_tools', 'vivarium_inputs']
     versions = {}
 
     for l in libraries:
         r = requests.get(f'https://pypi.org/pypi/{l}/json')
         versions[f'{l}_version'] = r.json()['info']['version']
 
     return versions
+
+
+def generate_template(repo_name, repo_path, repo_url):
+    extra = {'package_name': repo_name,
+             'ssh_url': repo_url}
+    extra.update(get_library_versions())
+
+    template_url = 'https://github.com/ihmeuw/vivarium_research_template.git'
+    cookiecutter(template_url,
+                 no_input=True,
+                 extra_context=extra,
+                 output_dir=str(repo_path.parent),
+                 overwrite_if_exists=True)
+
+
+def update_repository(repo_path):
```

Comment:
Can these fail? The first thing I'm thinking of is when a person with unconfigured git tries to push for the first time (though that isn't failure)

The push could fail if internet is down, I suppose